jupyter_client/multikernelmanager.py

```diff
@@ -315,3 +315,13 @@ def connect_hb(self, kernel_id, identity=None):
         =======
         stream : zmq Socket or ZMQStream
         """
+
+    def determine_kernel_id(self, **kwargs):
```

Comment:
Maybe simpler verb 'new_kernel_id'?

Reply:
I'm okay with changing the verb in the method name - please see next comment.